### PR TITLE
Implement L2Projector and tests

### PR DESCRIPTION
**Pull request by BaoTruong1**
## Description
This Pull Request implements L2Projector for Task 14

- Assembles the mass matrix M
- Assembles the load vector b
- Solve the system Mc = b to compute projection coefficients

Implementation Details
- Uses composite trapezoidal rule for numerical integration
- 
## Linked Issue
Fixes # 29

## Implementation Checklist
- [ ] My class inherits from `PolynomialBasis` (if applicable).
- [ ] I have implemented all abstract methods (`evaluate_basis`, `fit`, etc.).
- [ ] I have used `numpy` vectorization to avoid inefficient Python loops.
- [ ] Documentation strings (docstrings) are provided for all public methods.

### 1. Mathematical Integrity
- [ ] **Convergence:** Does the implementation maintain expected accuracy? (Checked for Runge's phenomenon or stability).
- [ ] **Edge Cases:** Does it handle single points or non-equally spaced nodes correctly?
- [ ] **Domain Mapping:** Does the class correctly map the input domain to the basis domain (e.g., [-1, 1] for Chebyshev)?

### 2. API & Style
- [ ] **Method Signatures:** Methods match the Abstract Base Class exactly.
- [ ] **Naming:** Follows `CamelCase` for classes and `snake_case` for functions/variables.
- [ ] **Clean Code:** No "magic numbers"; constants are named and explained.

### 3. Testing Quality
- [ ] **Coverage:** Unit tests cover typical usage and edge cases.
- [ ] **Precision:** Tests use `np.testing.assert_allclose` instead of `==`.
- [ ] **Validation:** There is a test proving the basis can exactly reconstruct a polynomial of its own degree.

## Screenshots / Plots

### CODE REVIEW

File: src/numanalysislib/approximation/l2_project.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        """
+        self.n_quad = n_quad
+
+    def _integrate(self, f: Callable[[np.ndarray], np.ndarray], a: float, b: float) -> float:
+        """
+        Numerically integrate f over [a, b] using the composite trapezoidal rule.
+        """
+        x = np.linspace(a, b, self.n_quad)
+        y = f(x)
+        return np.trapezoid(y, x)
```
Comment (mtezzele) [inline src/numanalysislib/approximation/l2_project.py]: use #24 once merged

File: src/numanalysislib/approximation/l2_project.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    def __init__(self, n_quad: int = 1000):
+        """
+        Args:
+            n_quad: Number of quadrature points used in the trapezoidal rule.
+        """
+        self.n_quad = n_quad
+
+    def _integrate(self, f: Callable[[np.ndarray], np.ndarray], a: float, b: float) -> float:
+        """
+        Numerically integrate f over [a, b] using the composite trapezoidal rule.
```
Comment (clara-armstrong) [inline src/numanalysislib/approximation/l2_project.py]: You should consider adding all arguments and returns in your documentation.

File: tests/test_l2_project.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        """
+        Project f(x) = 3 onto PowerBasis(1).
+        The exact coefficients should be [3, 0].
+        """
+        basis = PowerBasis(degree=1)
+        projector = L2Projector(n_quad=2000)
+        f = lambda x: 3.0 * np.ones_like(x)
+        coeffs = projector.project(basis, f)
+        expected = np.array([3.0, 0.0])
+        np.testing.assert_allclose(coeffs, expected, atol=1e-4)
```
Comment (clara-armstrong) [inline tests/test_l2_project.py:57]: This looks good, but consider having a stricter tolerance.

File: src/numanalysislib/approximation/l2_project.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            M_ij = ∫ phi_i(x) phi_j(x) dx
+        """
+        M = np.zeros((basis.n_dofs, basis.n_dofs), dtype=float)
+
+        for i in range(basis.n_dofs):
+            for j in range(basis.n_dofs):
+                integrand = lambda x, i=i, j=j: (
+                    basis.evaluate_basis(i, x) * basis.evaluate_basis(j, x)
+                )
+                M[i, j] = self._integrate(integrand, basis.a, basis.b)
```
Comment (clara-armstrong) [inline src/numanalysislib/approximation/l2_project.py]: The mass matrix could be computed differently since it is symmetric.
